File: server/router/internal_system/future_inventory/us004ProductAssociationDetail.go
package future_inventory

import (
	v1 "gin-vue-admin/api/v1/internal_system/future_inventory"
	"gin-vue-admin/middleware"

	"github.com/gin-gonic/gin"
)

func InitUs004ProductAssociationDetailRouter(Router *gin.RouterGroup) {
	Us004ProductAssociationDetailRouter := Router.Group("us004ProductAssociationDetail").Use(middleware.OperationRecord())
	{
		Us004ProductAssociationDetailRouter.POST("createUs004ProductAssociationDetail", v1.CreateUs004ProductAssociationDetail)             // 新建Us004ProductAssociationDetail
		Us004ProductAssociationDetailRouter.DELETE("deleteUs004ProductAssociationDetail", v1.DeleteUs004ProductAssociationDetail)           // 删除Us004ProductAssociationDetail
		Us004ProductAssociationDetailRouter.DELETE("deleteUs004ProductAssociationDetailByIds", v1.DeleteUs004ProductAssociationDetailByIds) // 批量删除Us004ProductAssociationDetail
		Us004ProductAssociationDetailRouter.PUT("updateUs004ProductAssociationDetail", v1.UpdateUs004ProductAssociationDetail)              // 更新Us004ProductAssociationDetail
		Us004ProductAssociationDetailRouter.GET("findUs004ProductAssociationDetail", v1.FindUs004ProductAssociationDetail)                  // 根据ID获取Us004ProductAssociationDetail
		Us004ProductAssociationDetailRouter.GET("getUs004ProductAssociationDetailList", v1.GetUs004ProductAssociationDetailList)            // 获取Us004ProductAssociationDetail列表
	}
}

File: web/src/api/us004ProductAssociationDetail.js
import service from '@/utils/request'

// @Tags Us004ProductAssociationDetail
// @Summary 创建Us004ProductAssociationDetail
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.Us004ProductAssociationDetail true "创建Us004ProductAssociationDetail"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /us004ProductAssociationDetail/createUs004ProductAssociationDetail [post]
export const createUs004ProductAssociationDetail = (data) => {
     return service({
         url: "/us004ProductAssociationDetail/createUs004ProductAssociationDetail",
         method: 'post',
         data
     })
 }


// @Tags Us004ProductAssociationDetail
// @Summary 删除Us004ProductAssociationDetail
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.Us004ProductAssociationDetail true "删除Us004ProductAssociationDetail"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /us004ProductAssociationDetail/deleteUs004ProductAssociationDetail [delete]
 export const deleteUs004ProductAssociationDetail = (data) => {
     return service({
         url: "/us004ProductAssociationDetail/deleteUs004ProductAssociationDetail",
         method: 'delete',
         data
     })
 }

// @Tags Us004ProductAssociationDetail
// @Summary 删除Us004ProductAssociationDetail
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除Us004ProductAssociationDetail"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /us004ProductAssociationDetail/deleteUs004ProductAssociationDetail [delete]
 export const deleteUs004ProductAssociationDetailByIds = (data) => {
     return service({
         url: "/us004ProductAssociationDetail/deleteUs004ProductAssociationDetailByIds",
         method: 'delete',
         data
     })
 }

// @Tags Us004ProductAssociationDetail
// @Summary 更新Us004ProductAssociationDetail
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.Us004ProductAssociationDetail true "更新Us004ProductAssociationDetail"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /us004ProductAssociationDetail/updateUs004ProductAssociationDetail [put]
 export const updateUs004ProductAssociationDetail = (data) => {
     return service({
         url: "/us004ProductAssociationDetail/updateUs004ProductAssociationDetail",
         method: 'put',
         data
     })
 }


// @Tags Us004ProductAssociationDetail
// @Summary 用id查询Us004ProductAssociationDetail
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.Us004ProductAssociationDetail true "用id查询Us004ProductAssociationDetail"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /us004ProductAssociationDetail/findUs004ProductAssociationDetail [get]
 export const findUs004ProductAssociationDetail = (params) => {
     return service({
         url: "/us004ProductAssociationDetail/findUs004ProductAssociationDetail",
         method: 'get',
         params
     })
 }


// @Tags Us004ProductAssociationDetail
// @Summary 分页获取Us004ProductAssociationDetail列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.PageInfo true "分页获取Us004ProductAssociationDetail列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /us004ProductAssociationDetail/getUs004ProductAssociationDetailList [get]
 export const getUs004ProductAssociationDetailList = (params) => {
     return service({
         url: "/us004ProductAssociationDetail/getUs004ProductAssociationDetailList",
         method: 'get',
         params
     })
 }